### PR TITLE
Add support for Android

### PR DESCRIPTION
**Pull request by atavism**

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -139,22 +141,30 @@ func (g *ReleaseManager) CheckForUpdate(p *Params) (res *Result, err error) {
...
+
+	if p.OS == OS.Android {
+		// if Android, stop here and return the full binary to the client
+		return fullResult(update), nil
+	}
 
-		return r, nil
+	// Looking for the asset thay matches the current app checksum.
+	var current *Asset
+	if current, err = g.lookupAssetWithChecksum(p.OS, p.Arch, p.Checksum); err != nil {
```
Comment (xiam) [inline server/server.go]: Can we instead modify `lookupAssetWithChecksum` so it returns an error when `p.OS` is `OS.Android`?

Comment (atavism) [inline server/server.go]: Thanks, @xiam! That's cleaner. `lookupAssetWithChecksum` just returns an error if the `os` is `OS.Android` now.

Comment (atavism) [inline server/server.go]: @xiam How does this look now? Want to get this out w/ the first Pro release if you have a second to look at it!

**Comment by xiam** [comment]
It looks good, we need to test it on stage, tho, and that includes releasing a binary in some special Github repo. I think we could do that after finishing desktop UI.

**Comment by atavism** [comment]
Ok, sounds good. I tested it on the staging server pretty thoroughly already. Would you be able to create that special repo (and make sure our staging server uses it instead of lantern) to quickly verify?

**Comment by atavism** [comment]
Hey @xiam, are you free to take a look at this still? I feel like we've done enough testing with staging already we can just deploy to production whenever

**Comment by xiam** [comment]
Please allow me some more time @atavism, we've just added a special to-do that includes complete testing for all our systems here and this would make this easier to test for everyone: https://github.com/getlantern/lantern/issues/4609

**Comment by xiam** [comment]
Thanks @atavism! I'm gonna merge this and test it with https://github.com/getlantern/autoupdate-server/pull/2, then push to production.

**Comment by atavism** [comment]
Sweet, @xiam. Thanks!

**Comment by atavism** [comment]
Can you let me know when this is live in production to test?

**Pull request merged**